packages/x-date-pickers/src/DateCalendar/DayCalendar.tsx

```diff
@@ -574,7 +574,7 @@ export function DayCalendar<TDate extends PickerValidDate>(inProps: DayCalendarP
             key={i.toString()}
             variant="caption"
             role="columnheader"
-            aria-label={utils.format(utils.addDays(startOfCurrentWeek, i), 'weekday')}
+            aria-label={utils.format(weekday, 'weekday')}
```

Comment:
No idea why we were re-doing the day shift here, but I spent some time finding out why my `aria-label` were not updated in the test whereas the UI was good in the doc :laughing:

Reply:
Nice optimization. 👍